### PR TITLE
fix ros2 zurich

### PR DESCRIPTION
**Pull request by Jacfomg**
Not sure if this is due to the new version or something went wrong with the other fix. Anyhow,

image

Zeno working again. It would be good to have it merged soon.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

File: src/qibolab/instruments/zhinst.py [hunk 1]
```diff
@@ -894,45 +894,49 @@ def measure_relax(self, exp, qubits, relaxation_time, acquisition_type):
...
-                        # We adjust for smearing and remove smearing/2 at the end
-                        exp.delay(
-                            signal=f"acquire{q}",
-                            time=self.smearing * NANO_TO_SECONDS,
+
+                    if i == 0:
+                        # Integration weights definition or load from the chip folder
+                        weights_file = (
+                            INSTRUMENTS_DATA_FOLDER
+                            / f"{self.chip}/weights/integration_weights_optimization_qubit_{q}.npy"
```
Comment (GabrielePalazzo) [inline src/qibolab/instruments/zhinst.py:907]:
```suggestion
                            / self.chip / "weights" / f"integration_weights_optimization_qubit_{q}.npy"
```

**Review by GabrielePalazzo — COMMENTED** [review]
Seems to work for me too.

**Comment by andrea-pasquale** [comment]
I can confirm that it works also when reading out multiple qubits.

**Review by andrea-pasquale — DISMISSED** [review]

**Comment by andrea-pasquale** [comment]
Lets try merge this as soon as possible.

**Review by GabrielePalazzo — APPROVED** [review]

**Comment by Jacfomg** [comment]
Thanks for the review @GabrielePalazzo and @andrea-pasquale, let's confirm the data and merge.

**Comment by andrea-pasquale** [comment]
Now the code works but the results are wrong when playing with multiple qubits:
http://login.qrccluster.com:9000/I3DUZrrIQgmmfr1sS3MkEA==/
Compared to the execution on a single qubit:
http://login.qrccluster.com:9000/11cuoo16SUWlo5F6g6cpfQ==/

**Comment by Jacfomg** [comment]
> Now the code works but the results are wrong when playing with multiple qubits: http://login.qrccluster.com:9000/I3DUZrrIQgmmfr1sS3MkEA==/ Compared to the execution on a single qubit: http://login.qrccluster.com:9000/11cuoo16SUWlo5F6g6cpfQ==/

I think this got solved as I ckecked and I was doing 

acquire1 with weight1
acquire2 with weight2
acquire1 with (weight2) changed to weight1
acquire2 with weight2

so by printing them I checked they are right again and the code compiles, still it would be nice to run on hardware.

**Comment by Jacfomg** [comment]
@stavros11 should be ready apart from a couple lines I didn't get into coverage due to it being the driver

**Review by stavros11 — APPROVED** [review]
Confirmed this fixes the unrolling #646. 

It would still be good if this discussion
> Now the code works but the results are wrong when playing with multiple qubits: http://login.qrccluster.com:9000/I3DUZrrIQgmmfr1sS3MkEA==/ Compared to the execution on a single qubit: http://login.qrccluster.com:9000/11cuoo16SUWlo5F6g6cpfQ==/

is resolved now that hardware is accessible @andrea-pasquale @Jacfomg (or anyone who is playing with IQM).

**Comment by andrea-pasquale** [comment]
> is resolved now that hardware is accessible @andrea-pasquale @Jacfomg (or anyone who is playing with IQM).

I can give it a try

**Comment by Jacfomg** [comment]
Let's wait until this afternoon before merging. I may get some feedback from ZI on how to increase the number of readouts.

**Comment by andrea-pasquale** [comment]
Btw, it seems fine now:
http://login.qrccluster.com:9000/IPgfWncERkibrzjhjQCZqg==/
http://login.qrccluster.com:9000/1DEAnUVaR_6qvDLp652GhA==/

**Comment by Jacfomg** [comment]
I will merge this and leave increasing the number of readouts for when then respond.

**Pull request merged**